backend/akr/src/main/java/fi/oph/akr/scheduled/ExpiringAuthorisationsEmailCreator.java

```diff
@@ -51,31 +51,31 @@ public void checkExpiringAuthorisations() {
 
       authorisationRepository
         .findExpiringAuthorisations(expiryBetweenStart, expiryBetweenEnd, previousReminderSentBefore)
-        .forEach(authorisationId -> {
-          if (hasEquivalentAuthorisationExpiringLater(authorisationId)) {
+        .forEach(authorisation -> {
+          if (hasEquivalentAuthorisationExpiringLater(authorisation)) {
             return;
           }
           try {
-            clerkEmailService.createAuthorisationExpiryEmail(authorisationId);
+            clerkEmailService.createAuthorisationExpiryEmail(authorisation.getId());
           } catch (Exception e) {
             LOG.error("Creation of authorisation expiry email failed", e);
           }
         });
     });
   }
 
-  private boolean hasEquivalentAuthorisationExpiringLater(final Long expiringAuthorisationId) {
-    final Authorisation expiringAuthorisation = authorisationRepository.getReferenceById(expiringAuthorisationId);
-    final Translator translator = expiringAuthorisation.getTranslator();
-    return translator
-      .getAuthorisations()
+  private boolean hasEquivalentAuthorisationExpiringLater(final Authorisation authorisation) {
+    final Translator translator = authorisation.getTranslator();
+    final List<Authorisation> matchingAuthorisations = authorisationRepository.findMatchingAuthorisations(
+      translator.getId(),
+      authorisation.getBasis(),
+      authorisation.getFromLang(),
+      authorisation.getToLang()
+    );
+
+    return matchingAuthorisations
       .stream()
-      .filter(a -> a.getId() != expiringAuthorisationId)
-      .anyMatch(a ->
-        a.getTermEndDate().isAfter(expiringAuthorisation.getTermEndDate()) &&
-        Objects.equals(expiringAuthorisation.getFromLang(), a.getFromLang()) &&
-        Objects.equals(expiringAuthorisation.getToLang(), a.getToLang()) &&
-        Objects.equals(expiringAuthorisation.getBasis(), a.getBasis())
-      );
+      .filter(a -> a.getId() != authorisation.getId())
+      .anyMatch(a -> a.getTermEndDate().isAfter(authorisation.getTermEndDate()));
```

Comment:
Eikö tuo filter ole turha, ja anyMatch päivävertailu voitaisiin tehdä kannassa. Ei estäne mergeä.

Reply:
Voishan siinä repositorimetodissa periaatteessa olla parametri tyyliin `LocalDate termEndDateAfter` tms. Mietin että olisi tarvittaessa yleiskäyttöisempi tällaisenaan ja jos mergeän näin.